File: .github/scripts/generate_docs.py
import re

def generate_markdown(source_file, output_file):
    with open(source_file, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find the entire switch block
    switch_match = re.search(r'switch\s*\(command\)\s*\{(.*?)default:', content, re.DOTALL)
    if not switch_match:
        print("Oops! I couldn't find the command switch block... Check the script for me babe? 🎀")
        return

    switch_body = switch_match.group(1)
    
    # 1. Matches one or more 'case' lines
    # 2. Captures everything until it hits the next 'case' or 'return true;'
    pattern = re.compile(r'((?:case\s+"/[^"]+":\s*\n\s*)+)(.*?)(?=case|return\s+true;)', re.DOTALL)
    
    commands = []
    for match in pattern.finditer(switch_body):
        case_block = match.group(1)
        logic_block = match.group(2).strip()
        
        # Extract commands from the case block
        cases = re.findall(r'case\s+"/([^"]+)"', case_block)
        
        # Clean up the logic: remove awaits, trailing semicolons, and fix newlines
        clean_logic = logic_block.replace('await ', '').rstrip(';')
        clean_logic = " ".join(clean_logic.split()) # Remove internal newlines/tabs
        
        commands.append({
            "command": f"/{cases[0]}",
            "aliases": ", ".join([f"`/{c}`" for c in cases[1:]]) if len(cases) > 1 else "None",
            "action": f"```{clean_logic}```"
        })

    # Build Markdown table
    md = "# Command Documentation\n\n"
    md += "| Command | Aliases | Action / Function |\n"
    md += "| :--- | :--- | :--- |\n"
    for cmd in commands:
        md += f"| **{cmd['command']}** | {cmd['aliases']} | {cmd['action']} |\n"

    with open(output_file, 'w', encoding='utf-8') as f:
        f.write(md)

if __name__ == "__main__":
    generate_markdown('client/ChatForm.Commands.cs', 'assets/docs/COMMANDS.md')

File: .github/scripts/GenerateDocs.cs
#:package Microsoft.CodeAnalysis.CSharp@4.12.0

using System;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;
using System.Text;

try
{
    var cwd = Directory.GetCurrentDirectory();
    var sourceFile = Path.Combine(cwd, "client", "ChatForm.Commands.cs");
    var outputFile = Path.Combine(cwd, "assets", "docs", "COMMANDS.md");

    if (!File.Exists(sourceFile))
        throw new FileNotFoundException($"Oops! I couldn't find the file at {sourceFile}... Check the folder for me babe? 🎀 ");

    var code = File.ReadAllText(sourceFile);
    var tree = CSharpSyntaxTree.ParseText(code);
    var root = tree.GetRoot();

    var commands = root.DescendantNodes()
        .OfType<SwitchStatementSyntax>()
        .SelectMany(s => s.Sections)
        .Select(section =>
        {
            var caseLabels = section.Labels
                .OfType<CaseSwitchLabelSyntax>()
                .Select(l => l.Value.ToString().Trim('"'))
                .ToList();

            if (!caseLabels.Any())
                return null;

            var actionLines = section.Statements
                .Select(stmt => stmt.ToString().Trim())
                .Where(text => !text.Contains("return true"))
                .Select(text =>
                {
                    string p = text;
                    p = p.Replace("await ", "").TrimEnd(';');

                    // --- HANDLE CONCATENATED STRINGS & PRINT ---
                    if (p.Contains("chatBox.AppendText"))
                    {
                        // 1. Get all content between quotes
                        var matches = Regex.Matches(p, @"""(.*?)""");
                        if (matches.Count > 0)
                        {
                            // 2. Combine them into one string
                            var combined = string.Join("", matches.Cast<Match>().Select(m => m.Groups[1].Value));

                            // 3. Split by the C# newline marker "\r\n"
                            var lines = combined.Split(new[] { "\\r\\n" }, StringSplitOptions.RemoveEmptyEntries);

                            // 4. Clean each line and wrap in <code> tags
                            var formattedLines = lines.Select(line =>
                            {
                                var cleanLine = line.Trim().Replace("{", "&#123;").Replace("}", "&#125;");
                                return $"<code>Print: \"{cleanLine}\"</code>";
                            });

                            // 5. Join these lines with <br> so they actually stack
                            return string.Join("<br>", formattedLines);
                        }
                    }

                    // --- FLATTEN WHITESPACE ---
                    p = Regex.Replace(p, @"\s+", " ").Trim();

                    // --- ESCAPE BRACES & CLEAN SYMBOLS ---
                    p = p.Replace("{", "&#123;").Replace("}", "&#125;");
                    p = p.Replace("$\"", "\"");

                    return $"<code>{p}</code>";
                })
                .ToList();

            return new
            {
                Command = $"/{caseLabels.First().TrimStart('/')}",
                Aliases = caseLabels.Skip(1).Any()
                    ? string.Join(", ", caseLabels.Skip(1).Select(c => $"`/{c.TrimStart('/')}`"))
                    : "None",
                Action = string.Join("<br>", actionLines)
            };
        })
        .Where(c => c != null)
        .ToList();

    var sb = new StringBuilder();
    sb.Append("# Command Documentation\r\n\r\n");
    sb.Append("| Command | Aliases | Action / Function |\r\n");
    sb.Append("| :--- | :--- | :--- |\r\n");

    foreach (var cmd in commands!)
        sb.Append($"| **{cmd.Command}** | {cmd.Aliases} | {cmd.Action} |\r\n");

    var outputDir = Path.GetDirectoryName(outputFile) ?? cwd;
    Directory.CreateDirectory(outputDir);
    File.WriteAllText(outputFile, sb.ToString(), Encoding.UTF8);

    Console.WriteLine($"Documentation generated at {outputFile}. You're doing amazing, sweetie! 💝");
}
catch (Exception ex)
{
    Console.WriteLine($"{ex.Message}");
}